tests/trainer/logging_/test_train_loop_logging.py

```diff
@@ -749,3 +746,36 @@ def validation_epoch_end(self, *_) -> None:
     train_data = DataLoader(RandomDataset(32, 64), batch_size=2)
     val_data = DataLoader(RandomDataset(32, 64), batch_size=2)
     trainer.fit(model, train_dataloaders=train_data, val_dataloaders=val_data)
+
+
+def test_no_batch_size_extraction_with_specifying_explictly(tmpdir):
+    batch_size = BoringModel().train_dataloader().batch_size + 1
+    fast_dev_run = 2
+    log_val = 7
+
+    class CustomBoringModel(BoringModel):
+        def on_before_batch_transfer(self, batch, *args, **kwargs):
+            # This is an ambiguous batch which have multiple potential batch sizes
+            if self.trainer.training:
+                batch = {"batch1": torch.randn(batch_size, 10), "batch2": batch}
```

Comment:
Hi, @rohitgr7  could we add more test cases to reflect users have mentioned in issue #10576 ?  This test case only captures a very ideal situation.

Reply:
well this test is just to check in what cases we should make a call to `extract_batch_size`. for the warning/error check we have tests here: https://github.com/PyTorchLightning/pytorch-lightning/blob/29797720cdbd74f5440c8e7186c82fe39b2002aa/tests/utilities/test_data.py#L18

Since I am not super familiar with torch-geometric, feel free to add some more here :)

Reply:
Resolving this, we can add more when we find new examples that do not work